### PR TITLE
refactor: improve search algorithm with isar full text search

### LINKED ISSUE #26: Improve search algorithm with isar full text search
(Note I did not do this in #22 (though perhaps I should've; willing to hear comment on that action))

- Look at the [recipe for a full text search](https://isar.dev/recipes/full_text_search.html) (made by [Isar](https://github.com/isar/isar)) and incorporate that into the app.

### PR DESCRIPTION
**Pull request by tycebrown**
I did very little here. [`IsarLinks`](https://isar.dev/links.html#isarlinks) does not support using [where clauses](https://isar.dev/queries.html#where-clauses) (which is what the [full-text search recipe in isar](https://isar.dev/recipes/full_text_search.html) uses), so there would be little benefit to [creating an index](https://isar.dev/recipes/full_text_search.html#basic-example), except possibly a speed boost, which would be at the cost of only being able to detect if the words in an item's value start with a search query word). 

So, in this branch, the only thing I did was replace my `_parseSearchStr()` with `Isar.splitWords()`.

closes #26

### CODE REVIEW

File: lib/view/list_widget.dart [hunk 1]
```diff
@@ -3,7 +3,7 @@ import 'dart:async';
 import 'package:collection/collection.dart';
 import 'package:lists/model/item.dart';
 import 'package:lists/model/list_model.dart';
-import 'package:flutter/material.dart';
+import 'package:flutter/material.dart' hide SearchBar;
```
Comment (Pertempto) [inline lib/view/list_widget.dart]: What is this change for?

Comment (tycebrown) [reply lib/view/list_widget.dart]: I believe the most recent version of flutter includes their own [`SearchBar`](https://api.flutter.dev/flutter/material/SearchBar-class.html) widget, the name of which conflicts with ours.

Comment (Pertempto) [reply lib/view/list_widget.dart]: It would be better to rename our widget so that it doesn't conflict with things from Flutter

Comment (tycebrown) [reply lib/view/list_widget.dart]: How about `SearchField`?

**Comment by Pertempto** [comment]
Sounds good!

File: lib/view/list_widget.dart (hunk 1 above)
Comment (Pertempto) [reply lib/view/list_widget.dart]: Sounds good

**Review by Pertempto — APPROVED** [review]

**Pull request merged**